### PR TITLE
Add domain='auto' to pf.dsp.normalize

### LINKED ISSUE #572: `domain='auto'` option for dsp.normalize?
## General

* pyfar version: 0.6.5

## Description

When using `pyfar.dsp.normalize()` with a FrequencyData object and default values, an error is thrown due to the default `domain='time'`. It would be nice to add an option `domain='auto'`, which checks if `signal` is a FrequencyData object.

## What I Did

```
import pyfar as pf
signal = pf.FrequencyData([1, 2, 3], [1, 2, 3])
pf.dsp.normalize(signal)
```

### PR DESCRIPTION
**Pull request by hoyer-a**
Added `domain='auto'` as default for normalization. 
Sets domain to ``'freq'`` for _FrequencyData_ and ``'time'`` for _TimeData_ and _Signal_

Closes #572

### CODE REVIEW

File: pyfar/dsp/dsp.py [hunk 1]
```diff
@@ -2234,8 +2241,14 @@ def normalize(signal, reference_method='max', domain='time',
         raise TypeError(("Input data has to be of type 'Signal', 'TimeData' "
                          "or 'FrequencyData'."))
 
-    if domain not in ('time', 'freq'):
-        raise ValueError("domain must be 'time' or 'freq'.")
+    if domain not in ('time', 'freq', 'auto'):
+        raise ValueError("domain must be 'time', 'freq' or 'auto.")
```
Comment (f-brinkmann) [inline pyfar/dsp/dsp.py:2245]:
```suggestion
        raise ValueError(f"domain must be 'time', 'freq' or 'auto' but is '{domain}'.")
```

**Review by f-brinkmann — APPROVED** [review]
Thanks. I only had one comment for improving the error message, but otherwise approved.

**Review by f-brinkmann — APPROVED** [review]

**Review by sikersten — DISMISSED** [review]

**Review by ahms5 — DISMISSED** [review]

**Comment by f-brinkmann** [comment]
@hoyer-a - we decided that it should be a hotfix. Can you please cherry-pick the commts and branch from main? If the tests pass, you can merge directly without furhter review...

**Comment by f-brinkmann** [comment]
Closed in favor of #649

**Pull request closed without merging**